src/hotspot/os_cpu/bsd_aarch64/os_bsd_aarch64.cpp

```diff
@@ -514,7 +514,39 @@ static inline void atomic_copy64(const volatile void *src, volatile void *dst) {
 
 extern "C" {
   int SpinPause() {
-    return 0;
+    // We don't use StubRoutines::aarch64::spin_wait stub in order to
+    // avoid a costly call to os::current_thread_enable_wx() on MacOS.
+    // We should return 1 if SpinPause is implemented, and since there
+    // will be a sequence of 11 instructions for NONE and YIELD and 12
+    // instructions for NOP and ISB, SpinPause will always return 1.
+    uint64_t br_dst;
+    const int instructions_per_case = 2;
+    int64_t off = VM_Version::spin_wait_desc().inst() * instructions_per_case * Assembler::instruction_size;
+
+    assert(VM_Version::spin_wait_desc().inst() >= SpinWait::NONE &&
+           VM_Version::spin_wait_desc().inst() <= SpinWait::YIELD, "should be");
+    assert(-1 == SpinWait::NONE,  "should be");
+    assert( 0 == SpinWait::NOP,   "should be");
+    assert( 1 == SpinWait::ISB,   "should be");
+    assert( 2 == SpinWait::YIELD, "should be");
```

Comment:
nit: these should all be "must be" in keeping with the existing HotSpot style
for that type of assert(). In specification language, "should" is optional and
"must" is a requirement.